### ISSUE TITLE
Download from bitbucket private repo in ansible-galaxy

### ISSUE THREAD

**Issue by adrien-barret** [cognifloyd/ansible-role-bitbucket-sources#4]
Hi,

I'm looking to solve a problem and your role like to do it, but it's not really flexible for my problem :1. 
So my question is about what you do and ansible-galaxy. Yes you got less check with Galaxy but did we got any possibility to pull from bitbucket cloud private repo with ansible-galaxy and setup ssh key, user and repo inside the requirement or isn't possible ?

**Comment by cognifloyd** [comment]
I don't understand. What do you need the role to do that it's not doing?

**Comment by adrien-barret** [comment]
in the way to use a private repo, I hoepd to use ansible-galaxy or a meta, ... not to use a big role to just download from a repo ?

**Comment by cognifloyd** [comment]
What file do you want to put it in? A playbook, a tasks file, a meta file,
or somewhere else? Will you use it in another role or directly in a
playbook?

And how do you want to use it?
For a dummy private repo, let's say bitbucket.com/example/dummy how would
your ideal repo download work? Can you show me some example yaml (please
specify which file your example is for)?

**Comment by cognifloyd** [comment]
Ah. Are you looking for a way to download roles from bitbucket using
ansible-galaxy?

**Comment by adrien-barret** [comment]
yup, with ansible-galaxy, but on a private repo into bitbucket, seems galaxy can't do that if it,s not a github ?

**Comment by cognifloyd** [comment]
I don't think Galaxy supports anything but GitHub. There might be a plugin
framework for Galaxy, but I doubt it. I have heard however that significant
improvements are coming to Galaxy, so maybe it will become more extensible.
Then we can revisit adding bitbucket support to Galaxy.

**Comment by adrien-barret** [comment]
Good to know, did you have any détails on theses next updates?
